### PR TITLE
Add support for in-UI authentication against Apple ID

### PR DESCRIPTION
**Pull request by chrismessina**
feat(ios-apps): add iOS App Search extension with in‑UI auth, resilient downloads, and screenshot fetching

- New iOS App Search extension with AI evals
- App Store scraping with high‑res screenshots; configurable allowed domains
- Inline Apple ID login + 2FA; centralized/auth-aware error handling; conditional App Store URL action
- ipatool validation + secure executable path checks; auto‑license for free apps
- More reliable downloads: file integrity verification, TLS/network retries, rate‑limited API (100ms) and downloads (50ms), null‑safe child process streams
- iTunes API improvements: migrate to node‑fetch v3, buffer handling fix, URL constants, add artistName, robust typing
- Logging & UX: structured logger, redacted/sanitized logs, improved toast styles for partial failures
- DX & cleanup: update React/TS deps, add tests/fixtures, remove duplicate directory and obsolete scripts/tests, lint/format, restore config‑example, README/EmptyView/TODO updates

## Screenshots

2025-08-16 (23 33 07) Dia

screenshot-1
screenshot-2
screenshot-3
screenshot-4
screenshot-5
screenshot-6

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used in the `README` are located outside the metadata folder if they were not generated with our [metadata tool](https://developers.raycast.com/basics/prepare-an-extension-for-store#how-to-use-it)

### CODE REVIEW

**Comment by chrismessina** [comment]
@greptile could you take a look?

**Comment by chrismessina** [comment]
@greptile I've removed the duplicate screenshots from the PR body.

File: extensions/ios-apps/src/components/forms/AppleLoginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    if (!email.includes("@")) {
+      setEmailError("Please enter a valid Apple ID");
+      return;
+    }
+
+    if (!password) {
+      // Password error is handled by Form.PasswordField's required validation
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/components/forms/AppleLoginForm.tsx:27]: The error says required is not a valid prop on Form.PasswordField (PasswordFieldProps). Raycast form fields don’t support a required prop; instead, you validate yourself and set the error prop to show an inline error.

I'm adding explicit validation + error handling for the password.

File: extensions/ios-apps/src/utils/formatting.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        symbol = "Mex$";
+        break;
+      // Add more currency codes as needed
+      default:
+        symbol = currency; // Use the currency code if no symbol is defined
+    }
+  }
+
+  return `${symbol}${price}`;
+}
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/utils/formatting.ts:54]: What changed:

- Replaced hardcoded symbol mapping with `Intl.NumberFormat`.
- Preserved “Free” for zero values (e.g., "0", "0.00").
- Added safe fallbacks for invalid/missing currency codes.
-

File: extensions/ios-apps/src/hooks/use-app-search.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+  // Create a debounced version of the search function that doesn't change on re-renders
+  const debouncedSearch = useCallback(
+    debounce((query: string) => {
+      // Wrap in a try/catch to handle errors locally
+      try {
+        performSearch(query);
+      } catch (err) {
+        handleSearchError(err);
+      }
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/hooks/use-app-search.ts:86]: Completed: redundant try/catch removed.

File: extensions/ios-apps/src/hooks/use-app-search.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  totalResults: number;
+  searchText: string;
+  setSearchText: (text: string) => void;
+}
+
+/**
+ * Hook for searching apps with debounced input
+ * @param initialSearchText Initial search text
+ * @param debounceMs Debounce time in milliseconds
+ * @param limit Maximum number of results to return
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/hooks/use-app-search.ts:20]: Documentation now accurately reflects the function parameters.

File: extensions/ios-apps/CONFIG.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+This document describes the configuration module (`src/config.ts`) that provides access to user preferences and default values for the iOS Apps extension.
+
+## Overview
+
+The configuration module exposes four main settings:
+
+- **maxDownloadTimeout** (ms) - Timeout for individual downloads
+- **maxStallTimeout** (ms) - Timeout when no progress is made  
+- **tempCleanupOnExit** (bool) - Whether to clean up temporary files on exit
+- **integrityVerification** (enum) - Level of file integrity verification (`basic` | `checksum` | `off`)
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/CONFIG.md:12]: Complete: documentation is now consistent with the code.

File: extensions/ios-apps/src/hooks/useAuthNavigation.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  pushLoginForm: (onSuccess?: () => void) => void;
+  push2FAForm: (sessionToken: string, onSuccess?: () => void) => void;
+  popToRoot: () => void;
+}
+
+export function useAuthNavigation(): AuthNavigationHelpers {
+  const { push, pop } = useNavigation();
+
+  const push2FAForm = useCallback(
+    (sessionToken: string, onSuccess?: () => void) => {
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/hooks/useAuthNavigation.tsx:17]: **Summary of Changes**

- Updated `AuthNavigationHelpers.push2FAForm` to drop sessionToken.
- Updated `useAuthNavigation.push2FAForm` implementation and login flow call site.
- Updated `handleAuthError` to match the new push2FAForm signature and removed placeholder token usage.
- Verified related components and compiled successfully.

File: extensions/ios-apps/src/hooks/useAuthNavigation.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+          }}
+        />,
+      );
+    },
+    [push, pop, push2FAForm],
+  );
+
+  const popToRoot = useCallback(() => {
+    pop();
+  }, [pop]);
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/hooks/useAuthNavigation.tsx:82]: Fix Applied: Proper pop-to-root

File: extensions/ios-apps/src/hooks/use-app-download.ts [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+                await ensureAuthenticated();
+                logger.log(`[useAppDownload] Auth OK after login (catch). Resuming download for ${name} (${bundleId})`);
+                await showToast({ style: Toast.Style.Animated, title: "Resuming download..." });
+                await handleDownload(
+                  downloadParams.bundleId,
+                  downloadParams.name,
+                  downloadParams.version,
+                  downloadParams.price,
+                  showHudMessages,
+                );
```
[the remark this answers is not in the document]
Comment (chrismessina) [reply extensions/ios-apps/src/hooks/use-app-download.ts:237]: The missing operationId parameter has been added to both recursive retry calls. The global lock will now be properly maintained during login and 2FA retry flows.

**Comment by chrismessina** [comment]
Superseded by #17630 after migrating changes and resolving conflicts.
